OgreMain/src/OgreAutoParamDataSource.cpp

```diff
@@ -463,6 +463,23 @@ namespace Ogre {
         return mCameraPositionObjectSpace;
     }
     //-----------------------------------------------------------------------------
+    const Vector4& AutoParamDataSource::getCameraPositionWorldSpace(void) const
+    {
+        if (mCameraPositionDirty)
+        {
+            Vector3 vec3 = mCurrentCamera->getDerivedPosition();
+            // We don't include the offset for camera relative rendering.
+            // This is intended for use when camera relative rendering is enabled
+            //   and the true world position is needed by a shader.
+            mCameraPosition[0] = vec3[0];
```

Comment:
this will overwrite the relative camera position. Please follow `getCameraPositionObjectSpace`

Reply:
just took a closer look at the file. Maybe it is even better to expose `ACT_CAMERA_RELATIVE_OFFSET` instead. We already have `mCameraRelativePosition` and it is used for various other `ACT_*` types.
This way one does not have to create `*_WORLD_SPACE` variants of those.

Reply:
that sounds great.  i like the idea of giving it a distinct name.  so rename ACT_CAMERA_POSITION_WORLD_SPACE to ACT_CAMERA_RELATIVE_POSITION (and coresponding functions, etc) and return mCameraRelativePosition.  (or did you have something else in mind with "OFFSET"?).

Reply:
ACT_CAMERA_RELATIVE_POSITION sounds fine